src/firewall/iptables.rs

```diff
@@ -64,6 +80,18 @@ impl firewall::FirewallDriver for IptablesDriver {
     // teardown_network should only be called in the case of
     // a complete teardown.
     fn teardown_network(&self, tear: TearDownNetwork) -> Result<(), Box<dyn Error>> {
+        let mut interface = "".to_string();
+        if let Some(_int) = tear.config.net.network_interface.clone() {
+            match tear.config.net.network_interface {
+                Some(iface) => interface = iface,
+                None => {
+                    return Err(Box::new(crate::error::NetavarkError {
+                        error: "failed to get interface".to_string(),
+                        errno: 1,
+                    }));
+                }
+            }
+        }
```

Comment:
same here, also we should avoid to error in cleanup and try to remove as much as possible.

Reply:
@Luap99 how does that work? I am unfamiliar with how rust errors work